lib/internal/http2/core.js

```diff
@@ -2337,7 +2339,10 @@ class Http2SecureServer extends TLSServer {
     super(options, connectionListener);
     this[kOptions] = options;
     this.timeout = kDefaultSocketTimeout;
-    this.on('newListener', setupCompat);
+    this.on('newListener', (ev) =>
+      setupCompat.call(this, this[kOptions].ServerRequest,
```

Comment:
I think it doesn't change anything but did you try using `Function.prototype.bind()`? It may be a little faster and easier to read.

Reply:
It doesn't even need to be bound here. `setupCompat` has access to this[kOptions].